<?xml version="1.0"?>
<launch>
    <!-- 노드 이름 설정 -->
    <arg name="node_name" default="cream_ioniq_node" />
    
    <!-- 설정 파일 경로 -->
    <arg name="config_file" default="$(find cream_ioniq)/config/lane_detection_config.yaml" />
    
    <!-- 사용할 노드 스크립트 선택 (기존 버전 또는 모듈형 버전) -->
    <arg name="use_modular" default="true" />
    
    <!-- 기존 모놀리식 노드 -->
    <group unless="$(arg use_modular)">
        <!-- 파라미터 파일 로드 (기존 버전용) -->
        <rosparam command="load" file="$(find cream_ioniq)/config/bev_params.yaml" ns="$(arg node_name)" />
        
        <node pkg="cream_ioniq" type="cream_ioniq_node.py" name="$(arg node_name)" output="screen">
        </node>
    </group>
    
    <!-- 새로운 모듈형 노드 -->
    <group if="$(arg use_modular)">
        <node pkg="cream_ioniq" type="cream_ioniq_node_modular.py" name="$(arg node_name)" output="screen">
            <!-- 설정 파일 경로를 파라미터로 전달 -->
            <param name="config_path" value="$(arg config_file)" />
            
            <!-- 프레임 ID 설정 -->
            <param name="frame_id" value="base_link" />
        </node>
    </group>
</launch> 